src/core/mac/mac.cpp

```diff
@@ -1241,13 +1299,70 @@ void Mac::HandleTransmitDone(TxFrame &aFrame, RxFrame *aAckFrame, otError aError
 
             if (mBroadcastTransmitCount < kTxNumBcast)
             {
-                mSubMac.Send();
+#if OPENTHREAD_CONFIG_MULTI_RADIO
+                {
+                    RadioTypes radioTypes;
+                    radioTypes.Add(kRadioTypeIeee802154);
+                    mLinks.Send(aFrame, radioTypes);
+                }
+#else
+                mLinks.Send();
+#endif
                 ExitNow();
             }
 
             mBroadcastTransmitCount = 0;
         }
     }
+#endif // #if OPENTHREAD_CONFIG_RADIO_LINK_IEEE_802_15_4_ENABLE
+
+#if OPENTHREAD_CONFIG_MULTI_RADIO
+    if (!aFrame.IsEmpty())
+    {
+        RadioType  radio          = aFrame.GetRadioType();
+        RadioTypes requriedRadios = mLinks.GetTxFrames().GetRequiredRadioTypes();
+
+        Get<RadioSelector>().UpdateOnSendDone(aFrame, aError);
```

Comment:
Could we consider just selecting radio when establish a link? Thus, Thread core doesn't need to keep track all radio supported by itself and peers, and we might have a simpler solution. And we can also make switching to a better radio as an optional feature just like the better parent feature.

Reply:
It is a possible different approach but I feel more limiting (in the longer run).
We need to somehow track which neighbor support which radio to enable radio switch recovery feature. Having it in core and single module `RadioSelector` is not that complex and  allows us to send over multiple radios either in parallel (e.g., for broadcast and/or probe mechanism to discover/track new supported radios) or sequentially (e.g. if a tx over a radio fails and there are other radio options supported by device and destination neighbor, we can consider retrying the message/frame on another radio).